### PR TITLE
Feat: solana family integration

### PR DESCRIPTION
**Pull request by RamyEB**
## Solana integration 

- [x] Created a new family 
- [x] Added tests

### CODE REVIEW

File: packages/core/src/families/solana/types.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import type { z } from "zod";
+import type { TransactionCommon } from "../index";
+import type { schemaRawSolanaTransaction } from "./validation";
+
+export interface SolanaTransaction extends TransactionCommon {
+  readonly family: RawSolanaTransaction["family"];
```
Comment (JunichiSugiura) [inline packages/core/src/families/solana/types.ts:6]: Missing transaction model field.

https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/families/solana/types.ts#L175

**Review by JunichiSugiura — CHANGES_REQUESTED** [review]

File: packages/core/src/families/solana/serializer.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+export function serializeSolanaTransaction({
+  family,
+  amount,
+  recipient,
+}: SolanaTransaction): RawSolanaTransaction {
+  return {
+    family,
+    amount: amount.toString(),
+    recipient,
```
Comment (cksanders) [inline packages/core/src/families/solana/serializer.ts:12]: model is required. https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/families/solana/transaction.ts#L34

File: packages/core/src/families/solana/serializer.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+export function deserializeSolanaTransaction({
+  family,
+  amount,
+  recipient,
+}: RawSolanaTransaction): SolanaTransaction {
+  return {
+    family,
+    amount: new BigNumber(amount),
+    recipient,
```
Comment (cksanders) [inline packages/core/src/families/solana/serializer.ts:24]: add model https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/families/solana/transaction.ts#L34

File: packages/core/tests/serializers.spec.ts [hunk 4]
```diff
@@ -520,6 +522,40 @@ describe("serializers.ts", () => {
...
     });
+
+    describe("solana", () => {
+      const family = schemaFamilies.enum.solana;
+
+      it("should succeed to serialize a solana transaction with resource and duration", () => {
+        const transaction: SolanaTransaction = {
+          family,
+          amount: new BigNumber(100),
+          recipient: "recipient",
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:533]: add model

File: packages/core/tests/serializers.spec.ts [hunk 5]
```diff
@@ -520,6 +522,40 @@ describe("serializers.ts", () => {
         });
       });
     });
+
+    describe("solana", () => {
+      const family = schemaFamilies.enum.solana;
+
+      it("should succeed to serialize a solana transaction with resource and duration", () => {
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:529]:
```suggestion
      it("should serialize a Solana transaction", () => {
```

File: packages/core/tests/serializers.spec.ts [hunk 6]
```diff
@@ -520,6 +522,40 @@ describe("serializers.ts", () => {
...
+        };
+        const serializedTransaction = serializeTransaction(transaction);
+
+        expect(serializedTransaction).toEqual({
+          family,
+          amount: "100",
+          recipient: "recipient",
+        });
+      });
+    });
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:558]: remove dup test

File: packages/core/tests/serializers.spec.ts [hunk 7]
```diff
@@ -977,6 +1013,42 @@ describe("serializers.ts", () => {
...
+
+        const transaction = deserializeTransaction(serializedTransaction);
+
+        expect(transaction).toEqual({
+          family,
+          amount: new BigNumber(100),
+          recipient: "recipient",
+        });
+      });
+    });
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:1051]: remove dup test

File: packages/core/tests/serializers.spec.ts [hunk 8]
```diff
@@ -977,6 +1013,42 @@ describe("serializers.ts", () => {
         });
       });
     });
+
+    describe("solana", () => {
+      const family = schemaFamilies.enum.solana;
+
+      it("should succeed to deserialize a solana transaction with resource and duration", () => {
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:1020]:
```suggestion
      it("should serialize an Elrond transaction", () => {
```

**Review by JunichiSugiura — CHANGES_REQUESTED** [review]
`npx changelog` to generate just in case. Then LGTM.

File: packages/core/src/families/solana/types.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+  model: object;
```
Comment (JunichiSugiura) [inline packages/core/src/families/solana/types.ts:7]: It should be typed.

File: packages/core/src/families/solana/validation.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
+import { z } from "zod";
+import { schemaFamilies, schemaTransactionCommon } from "../common";
+
+export const schemaRawSolanaTransaction = schemaTransactionCommon.extend({
+  family: z.literal(schemaFamilies.enum.solana),
+  model: z.string(),
```
Comment (JunichiSugiura) [inline packages/core/src/families/solana/validation.ts:6]: hmm, is this a stringified object?

**Review by JunichiSugiura — CHANGES_REQUESTED** [review]
Sorry, one more thing

File: packages/core/src/families/solana/validation.ts (hunk 10 above)
Comment (RamyEB) [reply packages/core/src/families/solana/validation.ts:6]: yep

**Review by cksanders — APPROVED** [review]

**Pull request merged**